icaruscode/TPC/ICARUSWireCell/icarus/wcls-multitpc-sim-drift-simchannel-yzsim-refactored.jsonnet

```diff
@@ -0,0 +1,495 @@
+      // - If nonzero, force number of ticks in output waveforms.
+      // - If zero, use whatever input data has. (default)
+      // - If -1, use value as per LS's detector properties service.
+      // nticks: params.daq.nticks,
+      // nticks: -1,
+      // chanmaskmaps: ['bad'],
+    },
+  }, nin=1, nout=1, uses=[duoanodes[n]])
+  for n in std.range(0,3)],
+
+  // The noise filtered "ADC" values.  These are truncated for
+  // art::Event but left as floats for the WCT SP.  Note, the tag
+  // "raw" is somewhat historical as the output is not equivalent to
+  // "raw data".
+  nf_digits: wcls.output.digits(name='nfdigits', tags=['raw']),
+
+  // The output of signal processing.  Note, there are two signal
+  // sets each created with its own filter.  The "gauss" one is best
+  // for charge reconstruction, the "wiener" is best for S/N
+  // separation.  Both are used in downstream WC code.
+  sp_signals: wcls.output.signals(name='spsignals', tags=['gauss', 'wiener']),
+
+  // save "threshold" from normal decon for each channel noise
+  // used in imaging
+  sp_thresholds: wcls.output.thresholds(name='spthresholds', tags=['threshold']),
+};
+
+//local deposio = io.numpy.depos(output);
+local drifter = sim.drifter;
+local setdrifter = g.pnode({
+            type: 'DepoSetDrifter',
+            data: {
+                drifter: "Drifter"
+            }
+        }, nin=1, nout=1,
+        uses=[drifter]);
+
+local localeLiftime = [std.extVar('lifetime') * wc.us,std.extVar('lifetime') * wc.us,std.extVar('lifetime') * wc.us,std.extVar('lifetime') * wc.us,std.extVar('lifetime') * wc.us,std.extVar('lifetime') * wc.us,std.extVar('lifetime') * wc.us,std.extVar('lifetime') * wc.us];
+
+local drifters = [{
+        local xregions = wc.unique_list(std.flattenArrays([v.faces for v in params.det.volumes])),
+
+        type: "Drifter",
+	name: "drifter%d" %n, //%std.floor(n/45),
+        data: params.lar {
+            rng: wc.tn(tools.random),
+            xregions: xregions,
+            time_offset: params.sim.depo_toffset,
+            drift_speed: params.lar.drift_speed,
+            fluctuate: params.sim.fluctuate,
+
+            DL: params.lar.DL,
+            DT: params.lar.DT,
+            lifetime: localeLiftime[std.floor(n/45)],
+	    charge_scale: 1 //std.mod(n,15)+1 //needs to be 1 
+        },
+    } 
+    for n in std.range(0,359)];  
+
+local setdrifters = [g.pnode({
+                     type: 'DepoSetDrifter',
+		     name: 'setdrifters%d' %n, 
+                     data: {
+		              drifter: wc.tn(drifters[n])
+			      #drifter: "Drifter"
+           		   }
+	             }, nin=1, nout=1,
+	             uses=[drifters[n]])
+                     #uses=[drifter])  
+		     for n in std.range(0,359)];
+
+local scalers = [{
+
+        type: "Scaler",
+	name: "scaler%d" %n, //%std.floor(n/45),
+        data: params.lar {
+	        	 yzmap_scale_filename: 'yzmap_gain_icarus_v2_run2.json',
+			 bin_width:  10*wc.cm,
```

Comment:
How does this handle depositions that are outside the binning of the YZ maps? I don't think I included the whole TPC when I made those.